### PR TITLE
More fixes

### LINKED ISSUE #13295: Elgg 3.3: "Inspect" - "Menus" broken
Fatal error
Argument 1 passed to Elgg\UserCapabilities::canEditAnnotation() must be an instance of ElggEntity, bool given, called in /path_to_Elgg_root/vendor/elgg/elgg/engine/classes/ElggAnnotation.php on line 104

For items in "register", "menu:annotation" there's no entity but an annotation the items belong to. The check in canEditAnnotation() expects an entity though.

### PR DESCRIPTION
**Pull request by jeabakker**
fixes #13295 and #13296

### CODE REVIEW

File: actions/login.php [hunk 1]
```diff
@@ -15,13 +15,23 @@
...
 
 // check if logging in with email address
-if (elgg_strpos($username, '@') !== false && ($users = get_user_by_email($username))) {
-	$username = $users[0]->username;
+if (elgg_strpos($username, '@') !== false) {
+	$users = elgg_call(ELGG_SHOW_DISABLED_ENTITIES, function() use ($username) {
+		return get_user_by_email($username);
+	});
+	
+	if (count($users) === 1) {
```
Comment (jdalsem) [inline actions/login.php:23]: why did you change this logic?

File: mod/uservalidationbyemail/start.php [hunk 2]
```diff
@@ -111,9 +111,10 @@ function uservalidationbyemail_check_manual_login(\Elgg\Event $event) {
 		if ($user->isEnabled() && $user->isValidated() !== false) {
 			return;
 		}
-		
-		if (elgg_get_plugin_user_setting('email_validated', $user->guid, 'uservalidationbyemail')) {
-			// email address already validated
+
+		$validated = elgg_get_plugin_user_setting('email_validated', $user->guid, 'uservalidationbyemail');
```
Comment (jdalsem) [inline mod/uservalidationbyemail/start.php:115]: you could also solve this with a default value... this way it is harder to read

**Review by jdalsem — CHANGES_REQUESTED** [review]

File: actions/login.php (hunk 1 above)
Comment (jeabakker) [reply actions/login.php:23]: security

File: languages/en.php [hunk 3]
```diff
@@ -120,6 +120,7 @@
 	'LoginException:ChangePasswordFailure' => 'Failed current password check.',
 	'LoginException:Unknown' => 'We could not log you in due to an unknown error.',
 	'LoginException:AdminValidationPending' => "Your account needs to be validated by a site administrator before you can use it. You'll be notified when your account is validated.",
+	'LoginException:DisabledUser' => "Your account has been disabled. You're not allowed to login",
```
[the remark this answers is not in the document]
Comment (jeabakker) [reply languages/en.php:123]: No the others are wrong ;)

File: mod/uservalidationbyemail/start.php (hunk 2 above)
Comment (jeabakker) [reply mod/uservalidationbyemail/start.php:115]: correct

File: actions/login.php (hunk 1 above)
Comment (jeabakker) [reply actions/login.php:23]: i'll change it back

**Review by jdalsem — APPROVED** [review]

**Pull request merged**